docs/DeveloperGuide.adoc

```diff
@@ -767,16 +842,117 @@ Use case ends.
 
-{More to be added}
+[discrete]
+=== Use case: Edit a specific tag word
+
+*MSS*
+
+1.  User requests to edit a tag
+2.  AddressBook shows the tag about to be edited
+3.  User requests to a new word to replace the current tag
+4.  AddressBook edits the tag word
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. The tag does not exist
++
+Use case ends.
+
+* 3b. The new word to replace the given tag is an existing tag
++
+[none]
+** 3b1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Type an incorrect command
+
+*MSS*
+
+1.  User requests with an incorrect command
+2.  AddressBook shows the suggested command based on user's misspelling or prompts help if the incorrect command cannot be identified
+3.  User requests with a correct command
++
+Use case ends.
+
+[discrete]
+=== Use case: Edit person
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to edit the details of the person identified by the index number in the listing
+4.  AddressBook edits the details of the identified person
+
++
+Use case ends.
+
+*Extensions*
+
+* 1a. AddressBook detects an error in the entered command
++
+[none]
+** 1a1. AddressBook requests for the correct command
++
+Use case resumes at step 2
+
+* 3a. AddressBook detects an error in the entered command
++
+[none]
+** 3a1. AddressBook requests for the correct command
++
+[none]
+** 3a2. User enters new command
++
+Use case resumes from step 4.
+
+[discrete]
+=== Use case: Undo
+
+*MSS*
+
+1.  User requests to undo previous action
+2.  AddressBook search for the previous action done
+3. AddressBook undo the previous action
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. The previous command is undoable.
++
+Use case resume at step 2.
+
+* 3a. All previous commands are undoable.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case ends.
+
+
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Have a user interface that is visible for use indoors as well as outdoors
+.  A user who is not very tech savvy will be able to familiar in using the application within 1 day of use
+.  Only the user himself can access his address book
+.  Should be able to easily find a person within a large amount of contacts in terms of robustness in searching capability (i.e. the use of multiple filters)
+.  Should be able to recover the address book data in case of loss of data
+.  User guide is clear and concise
+.  Should be as responsive as possible (a maximum of 0.3 millisecond)
```

Comment:
May not be realistic. In fact it depends on the amount of data. Also, seems arbitrary: why 0.3 ms and not 0.5ms?